mlir/lib/Dialect/MemRef/IR/MemRefOps.cpp

```diff
@@ -2353,6 +2353,16 @@ LogicalResult ExpandShapeOp::verify() {
            << " dynamic dims while output_shape has " << getOutputShape().size()
            << " values";
 
+  // Verify if provided output shapes are in agreement with output type.
+  DenseI64ArrayAttr staticOutputShapes = getStaticOutputShapeAttr();
+  ArrayRef<int64_t> resShape = getResult().getType().getShape();
+  unsigned staticShapeNum = 0;
+
+  for (unsigned i = 0, e = resShape.size(); i < e; ++i)
```

Comment:
You should be able to use a range loop (with `enumerate` if you want the `pos` in the error message)

Reply:
Most modern loop over indices would be this right now:

```suggestion
  for (int i = : llvm::seq<int>(0, resShape.size())
```

But the enumerate may even be nicer:

```suggestion
  for (auto [pos, shape] : llvm::enumerate(resShape))
```